### PR TITLE
fix(Tool/MMAP): Fixed invalid check causing certain tiles not to build

### PR DESCRIPTION
**Pull request by Kitzunu**
* cherry-pick commit (https://github.com/TrinityCore/TrinityCore/commit/e898b6e7942971a9bc96af54737ee971b99073a8)

Co-Authored-By: robinsch <robinsch@users.noreply.github.com>

## Changes Proposed:

example: mmaps_generator 230 will cause tile 33, 29 to not build because of the check. causing NPCs to not have mmap information in some parts of the map.

.go xyz 1380 -831 -88 230 has no mmap tile information because of above check.
https://i.imgur.com/Pypeu4f.png

## Issues Addressed:

- Closes 

## Tests Performed:

- 
- 

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Kitzunu** [comment]
Need to re-extract MMAPS

**Comment by pklloveyou** [comment]
Tested

**Comment by acidmanifesto** [comment]
hey. i am just a bit curious, but with our maps being a mix being based on a older extraction and calculation meathod then tc is, would this tc cherry pick even be necessary, let alone applies to us?

**Comment by Kitzunu** [comment]
same code that handles tiles I guess

**Comment by Nyeriah** [comment]
This PR has nothing to do with calculations. The file is exactly the same as we have right now. This addresses certain tiles not being built entirely.

File: src/common/Collision/Maps/MapDefines.h [hunk 1]
```diff
@@ -13,7 +13,7 @@
 #define SIZE_OF_GRIDS            533.3333f
 
 #define MMAP_MAGIC 0x4d4d4150   // 'MMAP'
-#define MMAP_VERSION 12
+#define MMAP_VERSION 14
```
Comment (Nyeriah) [inline src/common/Collision/Maps/MapDefines.h:16]: why 14 though?

**Comment by acidmanifesto** [comment]
At the moment i am not being able to experience the issue this pr describes to fix. pretty much what i did was use a hunter with a pet deployed. and did .go xyz 1380 -831 -88 230 and my pet moved over the spot several times depending on my orientation and position moved on and around that xyz.

**Comment by pklloveyou** [comment]
After updating the mmaps file, several new files appear,It should work

**Review by Winfidonarleyan — APPROVED** [review]

**Comment by pklloveyou** [comment]
Tested

**Comment by Nyeriah** [comment]
@Kitzunu I've removed the version update because we'll be merging several PRs that force reextract fairly soon, so I can merge this right now without worrying aaaaand we'll force people to reextract fairly soon so it shouldn't matter.

**Pull request merged**